### DISCUSSION TITLE
How to handle Rewrite to a Route Group in v13 App folder

### DISCUSSION THREAD

**Discussion by SharadKumar** [vercel/next.js#45924] [Help] [answered]
Lets say, I have two domains: 
- `https://country1.domain.com`
- `https://country2.domain.com`

And I have Nextjs v13.x app directory with Route Groups:
- `./app/(country1)/page.tsx`
- `./app/(country2)/page.tsx`

I want the domains to be mapped to route-groups, so that any content variations can be handled. Hence, I tried something like the following in `middleware.ts`:
```tsx
...
if (hostname.startsWith("country1.") || req.geo.country === "XX") {
    // Rewrite to the Country1 site
    return NextResponse.rewrite(new URL(`/country1${path}`, req.url))
}

// Otherwise, Rewrite to the Country2 site in `./app/(country2)/:path*`
  return NextResponse.rewrite(new URL(`/country2${path}`, req.url))
...

```

How to go about routing the related domain to the relevant route-group?
Appreciate your help with this.

**Comment by devjiwonchoi** [comment, marked as answer]
You still need a folder to be read as a path, since `/(country1)/page.tsx` is just `/` as a route path.

> Even though routes inside (marketing) and (shop) share the same URL hierarchy, you can create a different layout for each group by adding a layout.js file inside their folders.

In your case, try `/(country1)/[country1]/page.tsx`.

## Reference
- [Organize routes without affecting the URL path](https://nextjs.org/docs/app/building-your-application/routing/route-groups#organize-routes-without-affecting-the-url-path)
